app/scripts/components/common/featured-slider-section.tsx

```diff
@@ -67,6 +67,15 @@ function FeaturedSliderSection(props: FeaturedSliderSectionProps) {
 
   if (!featuredItems.length) return null;
 
+  // sorting featuredItems based on dateProperty in descending order
+  if (dateProperty) {
+    featuredItems.sort((itemA: StoryData | DatasetData, itemB: StoryData | DatasetData) => {
+      const pubDateOfItemA = new Date(itemA[dateProperty]);
+      const pubDateOfItemB = new Date(itemB[dateProperty]);
+      return pubDateOfItemB.getTime() - pubDateOfItemA.getTime();
+    });
+  }
```

Comment:
Mutating the original array can lead to a challenging situation in the future. (You can find many resources with the key sentence such as 'why mutate is bad in js') 

```suggestion
  // Disable no-mutating rule since we are mutating the copy of the array
  // eslint-disable-next-line fp/no-mutating-methods 
const sortedFeaturedItems = dateProperty? [...featuredItems].sort((itemA: StoryData | DatasetData, itemB: StoryData | DatasetData) => {
      const pubDateOfItemA = new Date(itemA[dateProperty]);
      const pubDateOfItemB = new Date(itemB[dateProperty]);
      return pubDateOfItemB.getTime() - pubDateOfItemA.getTime();
    }) : featuredItems;
  }
```

And use `sortedFeaturedItems` instead of `featuredItems ` at line 92.